dvc/repo/add.py

```diff
@@ -3,52 +3,75 @@
 import os
 import logging
 import colorama
+from six import string_types
 
 from dvc.repo.scm_context import scm_context
 from dvc.stage import Stage
 from dvc.utils import walk_files, LARGE_DIR_SIZE
 from dvc.exceptions import RecursiveAddingWhileUsingFilename
-
+from dvc.progress import Tqdm
 from . import locked
 
-
 logger = logging.getLogger(__name__)
 
 
 @locked
 @scm_context
-def add(repo, target, recursive=False, no_commit=False, fname=None):
+def add(
+    repo,
+    target_list,
+    recursive=False,
+    no_commit=False,
+    fname=None,
+    progress=False,
+):
     if recursive and fname:
         raise RecursiveAddingWhileUsingFilename()
 
-    targets = _find_all_targets(repo, target, recursive)
+    if isinstance(target_list, string_types):
+        target_list = [target_list]
 
-    if os.path.isdir(target) and len(targets) > LARGE_DIR_SIZE:
-        logger.warning(
-            "You are adding a large directory '{target}' recursively,"
-            " consider tracking it as a whole instead.\n"
-            "{purple}HINT:{nc} Remove the generated DVC-file and then"
-            " run {cyan}dvc add {target}{nc}".format(
-                purple=colorama.Fore.MAGENTA,
-                cyan=colorama.Fore.CYAN,
-                nc=colorama.Style.RESET_ALL,
-                target=target,
-            )
-        )
+    stages_list = []
+    with Tqdm(
+        total=len(target_list),
+        desc="Adding",
+        unit="file",
+        disable=not progress,
+        leave=True,
+    ) as pbar:
+        for target in target_list:
+            targets = _find_all_targets(repo, target, recursive)
+            pbar.total += len(targets) - 1
 
-    stages = _create_stages(repo, targets, fname)
+            if os.path.isdir(target) and len(targets) > LARGE_DIR_SIZE:
+                logger.warning(
+                    "You are adding a large directory '{target}' recursively,"
+                    " consider tracking it as a whole instead.\n"
+                    "{purple}HINT:{nc} Remove the generated DVC-file and then"
+                    " run {cyan}dvc add {target}{nc}".format(
+                        purple=colorama.Fore.MAGENTA,
+                        cyan=colorama.Fore.CYAN,
+                        nc=colorama.Style.RESET_ALL,
+                        target=target,
+                    )
+                )
 
-    repo.check_modified_graph(stages)
+            stages = _create_stages(repo, targets, fname, pbar=pbar)
 
-    for stage in stages:
-        stage.save()
+            repo.check_modified_graph(stages)
 
-        if not no_commit:
-            stage.commit()
+            for stage in stages:
+                stage.save()
 
-        stage.dump()
+                if not no_commit:
+                    stage.commit()
 
-    return stages
+                stage.dump()
+
+            stages_list += stages
+        pbar.bar_format = pbar.BAR_FMT_DEFAULT.replace("|{bar:10}|", " ")
```

Comment:
Could you explain what this is for please?

Reply:
just removing the progress bar bit upon completion as it seems unnecessary. would expect we need a separate PR later to actually fully sort out exactly what stats we want displayed upon completion so this line will change in future.

Reply:
Hi. On this, and maybe unrelated to this PR but, are we sure we want the progress bars to go away after they're done, in general? It's kind of nice to be able to see what happened in the past when the command is done, for example if you run a long `dvc add` and go away from your computer for a while. Just a thought.

Reply:
yes in general I think it's not worth removing progress bars as aggressively as we've currently done but in this instance it's only remving the bar bit and not the stats 

```
100%|███████████|Adding 10000/10000 [00:43<00:00, 229.00file/s]
```

becomes

```
100% Adding             10000/10000 [00:43<00:00, 229.00file/s]
```

Reply:
still could do with more tidying in a different PR I think: `<00:00` is useless; total size in bytes would be nice; "Adding" should be "Added;" etc...

Reply:
Good ideas. Feel fee to open a separate issue with these concerns (or find an existing ones to add your comments) if this is not the main purpose of the present PR.

Reply:
@casperdcl Honestly, this is a relly obscure piece of code. At the minimum we need to put a comment there.